fimpy/pipeline/alignment.py

```diff
@@ -154,3 +156,116 @@ def apply_shifts(dataset, output_dir=None, block_size=120, n_jobs=10, verbose=Fa
         for i_block, (_, new_block) in enumerate(new_dataset.slices(as_tuples=True))
     )
     return new_dataset.finalize()
+
+
+def align_2p_volume(
+    dataset,
+    output_dir=None,
+    reference=None,
+    n_frames_ref=10,
+    across_planes=None,
+    prefilter_sigma=3.3,
+    upsample_factor=10,
+    max_shift=15,
+    n_jobs=20,
+    verbose=True,
+):
+    """ Function for complete alignment of two-photon, planar acquired stack
+
+    :param dataset: input H5Dataset
+    :param output_dir: optional, output destination directory, subdirectory aligned will appear
+    :param reference: optional, reference to align to
+    :param n_frames_ref: number of frames to take as reference mean, if reference is being calculated
+    :param across_planes: bool, True by default if reference is not provided, whether to align across planes
+    :param prefilter_sigma: feature size to filter for better alignment. if < 0 no filtering will take place
+    :param upsample_factor: granularity of subpixel shift
+    :param max_shift: maximum shift allowed
+    :param n_jobs: number of parallel jobs
+    :return: reference to align dataset
+    """
+
+    # prepare the destination
+    new_dataset = EmptySplitDataset(
```

Comment:
if resolution is in the original dataset, it would be good to pass it here

Reply:
Should I add an optional input variable for the resolution (with None as default)?

Reply:
no, it doesn't make sense to insert it here, it should only be passed if the original dataset has it

Reply:
Ok, so this request is not clear to me. What does _if the original dataset has it_ means? I thought because fimpy is public it should be general.

Reply:
So at some point it was decided that the SplitDataset class optionally contains resolution and dimension order attributes
e.g.
https://github.com/portugueslab/split_dataset/blob/1faa37954a1fe7bcd23021e2f93238f7b94affb5/split_dataset/split_dataset.py#L113
it makes sense that all operations that work on SplitDataset preserve this information. (so if you e.g. align a dataset which has these fields, the aligned dataset will have these attributes as well). A more elegant, consistent solution for all of this would use [xarray](http://xarray.pydata.org/en/stable/index.html), but @vigji is still ambivalent over file formats supported by xarray.

Reply:
I am ambivalent indeed :D so so far we are just storing the resolution, and if one day we move to xarray supporting formats we have the information to move there the stack easily

Reply:
it's not just resolution, dimension order is important too

Reply:
Can I merge the branches and deal with this later? I would like to start running alignment of new data